src/main/java/org/gooru/insights/services/ItemServiceImpl.java

```diff
@@ -391,6 +411,116 @@ private String appendQuery(String requestId, String prefix, List<Map<String, Obj
+			offSet = paginationDTO.getOffset();
+			paginationDTO.setLimit(defaultLimit);
+			requestParamsDTO.setPagination(paginationDTO);
+			
+			if (userMap == null) {
+				userMap = getUserObjectData(traceId,sessionToken);
+			}
+
+			Map<String, Boolean> checkPoint = getBaseAPIService().checkPoint(requestParamsDTO);
+			requestParamsDTO = getUserService().validateUserRole(traceId,requestParamsDTO, userMap);
+			String[] indices = getBaseAPIService().getIndices(requestParamsDTO.getDataSource().toLowerCase());
+			
+			do {
+				responseDTO = getEsService().generateQuery(traceId,requestParamsDTO, indices, checkPoint);
+				getCSVFileWriterService().generateCSVReport(new HashSet<String>(Arrays.asList(requestParamsDTO.getFields().split(APIConstants.COMMA))), responseDTO.getContent(), absoluteFilePath, delimiter, isNewFile);
+				checkPoint.put(Hasdatas.HAS_MULTIGET.check(), false);
+				/*Incrementing offset values */
+				offSet += defaultLimit;
+				paginationDTO.setOffset(offSet);
+				requestParamsDTO.setPagination(paginationDTO);
+				
+				checkPoint = getBaseAPIService().checkPoint(requestParamsDTO);
+				totalRows = Integer.valueOf(responseDTO.getPaginate().get(APIConstants.TOTAL_ROWS).toString());
+				isNewFile = false;
+			} while(offSet < totalRows);
+			
+		} catch (Exception e) {
+			logger.error("Error while writting file. {}", e);
+		}
+	}
+	
+	@Override
+	public ResponseParamDTO<Map<String, Object>> exportReport(HttpServletResponse response, String traceId, String data, String sessionToken, Map<String, Object> userMap) {
+
+		int maxLimit = Integer.valueOf(getBaseConnectionService().getExportReportCache().get(APIConstants.MAX_LIMIT));
+		int totalRows = 0;
+		String absoluteFilePath = null;
+		String fileName = UUID.randomUUID().toString();
+		ResponseParamDTO<Map<String, Object>> responseDTO = null;
+		
+		try {
+			
+			RequestParamsDTO requestParamsDTO = getBaseAPIService().buildRequestParameters(data);
+			
+			if (userMap == null) {
+				userMap = getUserObjectData(traceId,sessionToken);
+			}
+
+			Map<String, Boolean> checkPoint = getBaseAPIService().checkPoint(requestParamsDTO);
+			requestParamsDTO = getUserService().validateUserRole(traceId,requestParamsDTO, userMap);
+			String[] indices = getBaseAPIService().getIndices(requestParamsDTO.getDataSource().toLowerCase());
+
+			totalRows = Integer.valueOf(getEsService().generateQuery(traceId,requestParamsDTO, indices, checkPoint).getPaginate().get(APIConstants.TOTAL_ROWS).toString());
+			absoluteFilePath = getBaseConnectionService().getRealRepoPath().concat(fileName).concat(APIConstants.DOT).concat(APIConstants.CSV_EXTENSION);
+			
+			Map<String, Object> status = new HashMap<String, Object>();
+			final String resultLink = getBaseConnectionService().getAppRepoPath().concat(fileName).concat(APIConstants.DOT).concat(APIConstants.CSV_EXTENSION);
+			responseDTO = new ResponseParamDTO<Map<String,Object>>();
+			
+			if(totalRows > maxLimit) {
+				final String traceID = traceId;
```

Comment:
// final String traceID = traceId;
 Why we need to  reassign traceId in another variable ?